cmake/onnxruntime.cmake

```diff
@@ -228,13 +228,22 @@ target_link_libraries(onnxruntime PRIVATE
 )
 
 set_property(TARGET onnxruntime APPEND_STRING PROPERTY LINK_FLAGS ${ONNXRUNTIME_SO_LINK_FLAG} ${onnxruntime_DELAYLOAD_FLAGS})
-set_target_properties(onnxruntime PROPERTIES
-  PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
-  LINK_DEPENDS ${SYMBOL_FILE}
-  VERSION ${ORT_VERSION}
-  FOLDER "ONNXRuntime"
-)
-
+if (WIN32)
```

Comment:
@pranavsharma , what do you think? This change will change the DLL's name for debug version. I would consider it as a breaking change.  People who have this need can set https://cmake.org/cmake/help/latest/variable/CMAKE_DEBUG_POSTFIX.html without modifying ONNX Runtime's code.

Reply:
About build.bat with vs 2022 https://github.com/microsoft/onnxruntime/issues/20809#issuecomment-2143920841

Reply:
Was the issue resolved?

Reply:
Yes and I can use 
cmake --build . --target install  --config release 
or
cmake --build . --target install  --config debug
but dll name are identical this PR solve this https://github.com/microsoft/onnxruntime/issues/20387


Now with this PR I can import onnxruntime in my own c++ project using this CMakelist.txt. In vs2022 release dll onnxruntime.dll and debug dll onnxruntime_d.dll (idem for lib and provider)

```
cmake_minimum_required(VERSION 3.2)
set(projectName "onnxsample")

PROJECT (${projectName})

find_package(onnxruntime REQUIRED CONFIG)

if(onnxruntime_FOUND)
  get_property(_loc TARGET onnxruntime::onnxruntime PROPERTY LOCATION)
  message(STATUS "Found onnxruntime: ${_loc} (found version ${onnxruntime_VERSION})")
  foreach(prop IN ITEMS
      IMPORTED_LOCATION_DEBUG IMPORTED_LOCATION_RELEASE
      IMPORTED_IMPLIB_DEBUG IMPORTED_IMPLIB_RELEASE)
    get_target_property(_item_loc onnxruntime::onnxruntime ${prop})
	message(STATUS "Property ${prop} : ${_item_loc}")
    if(_item_loc)
      get_filename_component(_item_loc ${_item_loc} DIRECTORY)
      list(APPEND _locations ${_item_loc})
    endif()
  endforeach()
endif()

file(GLOB SourceFile
    "*.h"
    "*.cpp")
ADD_EXECUTABLE (${projectName} ${SourceFile})
if (onnxruntime_FOUND)
	include_directories(${onnxruntime_INCLUDE_DIRS})
	target_link_libraries (${projectName} LINK_PUBLIC  onnxruntime::onnxruntime)
else (onnxruntime_FOUND)
    message("A lib is not found")
    message("onnxruntime ${onnxruntime_FOUND}")
endif (onnxruntime_FOUND)

```

Reply:
What if you build it with 
```
--cmake_extra_defines CMAKE_DEBUG_POSTFIX "_d"
```
Would it work?